lib/CL/devices/tbb/tbb.c

```diff
@@ -286,13 +286,28 @@ void pocl_tbb_notify(cl_device_id device, cl_event event, cl_event finished) {
   cl_bool wake_thread = CL_FALSE;
   _cl_command_node *node = event->command;
 
-  if (finished->status < CL_COMPLETE) {
-      pocl_update_event_failed_locked (event);
-      return;
+  if (finished->status < CL_COMPLETE)
+  {
+    /* Unlock the finished event in order to prevent a lock order violation
```

Comment:
I see this part duplicated and I start to think that could we refactor some of the common event handling to the PoCL core side to simplify the drivers and avoid the duplication?